### PR TITLE
Refactor logging statements in recommender:checkpoint,input

### PR DESCRIPTION
**Pull request by omerap12**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
To align the logging with the best practices and guidelines established within the Kubernetes ecosystem.

#### Which issue(s) this PR fixes:

As part of https://github.com/kubernetes/autoscaler/issues/6905, I'm splitting this into multiple PRs rather than submitting a single large PR.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
NONE
```

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_source.go [hunk 1]
```diff
@@ -114,10 +114,10 @@ func (s *externalMetricsClient) List(ctx context.Context, namespace string, opts
 					return nil, err
 				}
 				if m == nil || len(m.Items) == 0 {
-					klog.V(4).Infof("External Metrics Query for VPA %s: resource %+v, metric %+v, No items,", klog.KRef(vpa.ID.Namespace, vpa.ID.VpaName), resourceName, metricName)
+					klog.V(4).InfoS("External Metrics Query for VPA: No items","vpa", klog.KRef(vpa.ID.Namespace, vpa.ID.VpaName),"resource", resourceName,"metric", metricName)
```
Comment (adrianmoisey) [inline vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_source.go:117]: Style:
```suggestion
					klog.V(4).InfoS("External Metrics Query for VPA: No items", "vpa", klog.KRef(vpa.ID.Namespace, vpa.ID.VpaName), "resource", resourceName, "metric", metricName)
```

File: vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_source.go [hunk 2]
```diff
@@ -114,10 +114,10 @@ func (s *externalMetricsClient) List(ctx context.Context, namespace string, opts
...
 					continue
 				}
-				klog.V(4).Infof("External Metrics Query for VPA %s: resource %+v, metric %+v, %d items, item[0]: %+v", klog.KRef(vpa.ID.Namespace, vpa.ID.VpaName), resourceName, metricName, len(m.Items), m.Items[0])
+				klog.V(4).InfoS("External Metrics Query for VPA","vpa", klog.KRef(vpa.ID.Namespace, vpa.ID.VpaName),"resource", resourceName,"metric", metricName,"itemCount", len(m.Items),"firstItem", m.Items[0])
```
Comment (adrianmoisey) [inline vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_source.go:120]: Style:
```suggestion
				klog.V(4).InfoS("External Metrics Query for VPA", "vpa", klog.KRef(vpa.ID.Namespace, vpa.ID.VpaName), "resource", resourceName, "metric", metricName, "itemCount", len(m.Items), "firstItem", m.Items[0])
```

File: vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_source.go (hunk 1 above)
Comment (omerap12) [reply vertical-pod-autoscaler/pkg/recommender/input/metrics/metrics_source.go:117]: yeap

**Comment by omerap12** [comment]
@adrianmoisey fixed, do you want me to rebase?

**Comment by adrianmoisey** [comment]
> @adrianmoisey fixed, do you want me to rebase?

Nah, I think it's fine.

File: vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go [hunk 3]
```diff
@@ -255,48 +255,48 @@ func (feeder *clusterStateFeeder) InitFromCheckpoints() {
...
+		klog.V(3).InfoS("Fetching checkpoints", "namespace", namespace)
 		checkpointList, err := feeder.vpaCheckpointClient.VerticalPodAutoscalerCheckpoints(namespace).List(context.TODO(), metav1.ListOptions{})
 		if err != nil {
-			klog.Errorf("Cannot list VPA checkpoints from namespace %s. Reason: %+v", namespace, err)
+			klog.ErrorS(err, "Cannot list VPA checkpoints", "namespace", namespace)
 		}
 		for _, checkpoint := range checkpointList.Items {
 
-			klog.V(3).Infof("Loading VPA %s/%s checkpoint for %s", checkpoint.ObjectMeta.Namespace, checkpoint.Spec.VPAObjectName, checkpoint.Spec.ContainerName)
+			klog.V(3).InfoS("Loading VPA checkpoint", "namespace", checkpoint.ObjectMeta.Namespace, "VPA", checkpoint.Spec.VPAObjectName, "container", checkpoint.Spec.ContainerName)
```
Comment (kwiesmueller) [inline vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go:265]: nit: consistency with other log lines where it's lowercase vpa
```suggestion
			klog.V(3).InfoS("Loading VPA checkpoint", "namespace", checkpoint.ObjectMeta.Namespace, "vpa", checkpoint.Spec.VPAObjectName, "container", checkpoint.Spec.ContainerName)
```

**Review by kwiesmueller — CHANGES_REQUESTED** [review]

**Comment by omerap12** [comment]
> @omerap12 can you address the remaining comments here?

Of course, I will adjust the code later

**Comment by omerap12** [comment]
adjusted @adrianmoisey

File: vertical-pod-autoscaler/pkg/recommender/checkpoint/checkpoint_writer.go [hunk 4]
```diff
@@ -108,11 +108,9 @@ func (writer *checkpointWriter) StoreCheckpoints(ctx context.Context, now time.T
 			}
 			err = api_util.CreateOrUpdateVpaCheckpoint(writer.vpaCheckpointClient.VerticalPodAutoscalerCheckpoints(vpa.ID.Namespace), &vpaCheckpoint)
 			if err != nil {
-				klog.Errorf("Cannot save VPA %s/%s checkpoint for %s. Reason: %+v",
-					vpa.ID.Namespace, vpaCheckpoint.Spec.VPAObjectName, vpaCheckpoint.Spec.ContainerName, err)
+				klog.ErrorS(err, "Cannot save VPA checkpoint", "namespace", vpa.ID.Namespace, "vpa", vpaCheckpoint.Spec.VPAObjectName, "container", vpaCheckpoint.Spec.ContainerName)
```
Comment (voelzmo) [inline vertical-pod-autoscaler/pkg/recommender/checkpoint/checkpoint_writer.go:111]: Let's simplify with `klog.KRef` and make clear that the reference is pointing to the VPA, not the checkpoint

```suggestion
				klog.ErrorS(err, "Cannot save checkpoint for VPA", klog.KRef(vpa.ID.Namespace, vpaCheckpoint.Spec.VPAObjectName), "container", vpaCheckpoint.Spec.ContainerName)
```

File: vertical-pod-autoscaler/pkg/recommender/checkpoint/checkpoint_writer.go [hunk 5]
```diff
@@ -108,11 +108,9 @@ func (writer *checkpointWriter) StoreCheckpoints(ctx context.Context, now time.T
...
 			} else {
-				klog.V(3).Infof("Saved VPA %s/%s checkpoint for %s",
-					vpa.ID.Namespace, vpaCheckpoint.Spec.VPAObjectName, vpaCheckpoint.Spec.ContainerName)
+				klog.V(3).InfoS("Saved VPA checkpoint", "namespace", vpa.ID.Namespace, "vpa", vpaCheckpoint.Spec.VPAObjectName, "container", vpaCheckpoint.Spec.ContainerName)
```
Comment (voelzmo) [inline vertical-pod-autoscaler/pkg/recommender/checkpoint/checkpoint_writer.go:113]: same here

```suggestion
				klog.V(3).InfoS("Saved checkpoint for VPA", klog.KRef(vpa.ID.Namespace, vpaCheckpoint.Spec.VPAObjectName), "container", vpaCheckpoint.Spec.ContainerName)
```

File: vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go [hunk 6]
```diff
@@ -128,7 +128,7 @@ func WatchEvictionEventsWithRetries(kubeClient kube_client.Interface, observer o
 			watchEvictionEventsOnce()
 			// Wait between attempts, retrying too often breaks API server.
 			waitTime := wait.Jitter(evictionWatchRetryWait, evictionWatchJitterFactor)
-			klog.V(1).Infof("An attempt to watch eviction events finished. Waiting %v before the next one.", waitTime)
+			klog.V(1).InfoS("An attempt to watch eviction events finished", "waiting", waitTime)
```
Comment (voelzmo) [inline vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go:131]:
```suggestion
			klog.V(1).InfoS("An attempt to watch eviction events finished", "waitTime", waitTime)
```

File: vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go [hunk 7]
```diff
@@ -255,48 +255,48 @@ func (feeder *clusterStateFeeder) InitFromCheckpoints() {
...
+		klog.V(3).InfoS("Fetching checkpoints", "namespace", namespace)
 		checkpointList, err := feeder.vpaCheckpointClient.VerticalPodAutoscalerCheckpoints(namespace).List(context.TODO(), metav1.ListOptions{})
 		if err != nil {
-			klog.Errorf("Cannot list VPA checkpoints from namespace %s. Reason: %+v", namespace, err)
+			klog.ErrorS(err, "Cannot list VPA checkpoints", "namespace", namespace)
 		}
 		for _, checkpoint := range checkpointList.Items {
 
-			klog.V(3).Infof("Loading VPA %s/%s checkpoint for %s", checkpoint.ObjectMeta.Namespace, checkpoint.Spec.VPAObjectName, checkpoint.Spec.ContainerName)
+			klog.V(3).InfoS("Loading VPA checkpoint", "namespace", checkpoint.ObjectMeta.Namespace, "vpa", checkpoint.Spec.VPAObjectName, "container", checkpoint.Spec.ContainerName)
```
Comment (voelzmo) [inline vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go:265]: same here.

```suggestion
			klog.V(3).InfoS("Loading checkpoint for VPA", klog.KRef(checkpoint.ObjectMeta.Namespace, checkpoint.Spec.VPAObjectName), "container", checkpoint.Spec.ContainerName)
```

File: vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go [hunk 8]
```diff
@@ -255,48 +255,48 @@ func (feeder *clusterStateFeeder) InitFromCheckpoints() {
...
+			klog.ErrorS(err, "Cannot list VPA checkpoints", "namespace", namespace)
 		}
 		for _, checkpoint := range checkpointList.Items {
 			vpaID := model.VpaID{Namespace: checkpoint.Namespace, VpaName: checkpoint.Spec.VPAObjectName}
 			_, exists := feeder.clusterState.Vpas[vpaID]
 			if !exists {
 				err = feeder.vpaCheckpointClient.VerticalPodAutoscalerCheckpoints(namespace).Delete(context.TODO(), checkpoint.Name, metav1.DeleteOptions{})
 				if err == nil {
-					klog.V(3).Infof("Orphaned VPA checkpoint cleanup - deleting %v/%v.", namespace, checkpoint.Name)
+					klog.V(3).InfoS("Orphaned VPA checkpoint cleanup - deleting", "namespace", namespace, "checkpoint", checkpoint.Name)
```
Comment (voelzmo) [inline vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go:297]:
```suggestion
					klog.V(3).InfoS("Orphaned VPA checkpoint cleanup - deleting", klog.KRef(namespace, checkpoint.Name))
```

File: vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go [hunk 9]
```diff
@@ -255,48 +255,48 @@ func (feeder *clusterStateFeeder) InitFromCheckpoints() {
...
 				} else {
-					klog.Errorf("Cannot delete VPA checkpoint %v/%v. Reason: %+v", namespace, checkpoint.Name, err)
+					klog.ErrorS(err, "Orphaned VPA checkpoint cleanup - error deleting", "namespace", namespace, "checkpoint", checkpoint.Name)
```
Comment (voelzmo) [inline vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go:299]:
```suggestion
					klog.ErrorS(err, "Orphaned VPA checkpoint cleanup - error deleting", klog.KRef(namespace, checkpoint.Name))
```

File: vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go [hunk 10]
```diff
@@ -318,27 +318,27 @@ func selectsRecommender(selectors []*vpa_types.VerticalPodAutoscalerRecommenderS
...
 			if !selectsRecommender(vpaCRD.Spec.Recommenders, &feeder.recommenderName) {
-				klog.V(6).Infof("Ignoring vpaCRD %s as current recommender's name %v doesn't appear among its recommenders", klog.KObj(vpaCRD), feeder.recommenderName)
+				klog.V(6).InfoS("Ignoring vpaCRD as current recommender's name doesn't appear among its recommenders", "vpaCRD", klog.KObj(vpaCRD), "recommenderName", feeder.recommenderName)
 				continue
 			}
 		}
 
 		if slices.Contains(feeder.ignoredNamespaces, vpaCRD.ObjectMeta.Namespace) {
-			klog.V(6).Infof("Ignoring vpaCRD %s in namespace %s as namespace is ignored", klog.KObj(vpaCRD), vpaCRD.Namespace)
+			klog.V(6).InfoS("Ignoring vpaCRD in namespace as namespace is ignored", "vpaCRD", klog.KObj(vpaCRD), "namespace", vpaCRD.Namespace)
```
Comment (voelzmo) [inline vertical-pod-autoscaler/pkg/recommender/input/cluster_feeder.go:341]: namespace is already part of the VPA object, we can simplify here

```suggestion
			klog.V(6).InfoS("Ignoring vpaCRD as this namespace is ignored", "vpaCRD", klog.KObj(vpaCRD))
```

**Review by voelzmo — CHANGES_REQUESTED** [review]
This is a great change, thanks for the adaptations! Some small comments where we can simplify the logs.

Also: there's a few occasions of `klog.Warnf`, could you also take care of those, while we're at it? Thanks!

**Comment by omerap12** [comment]
> This is a great change, thanks for the adaptations! Some small comments where we can simplify the logs.
> 
> Also: there's a few occasions of `klog.Warningf`, could you also take care of those, while we're at it? Thanks!

Thanks for your review! Ill just pinned [this](https://github.com/kubernetes/autoscaler/pull/7294#issuecomment-2422423460) ( no need to discuss things twice 😄 )

**Comment by voelzmo** [comment]
/lgtm
/approve

Thanks!

**Pull request merged**